{"uid":"989b70b0033b5e6c","name":"test_github","fullName":"tests.test_selene#test_github","historyId":"5167f7c71535600a353c42dca9c39857","time":{"start":1688982500824,"stop":1688982507546,"duration":6722},"status":"failed","statusMessage":"selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 4s, while waiting for:\nbrowser.element(('link text', 'eroshenkoam/allure-example')).click\n\nReason: NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"link text\",\"selector\":\"eroshenkoam/allure-example\"}\n  (Session info: headless chrome=114.0.5735.198); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\n#0 0x562055d354e3 <unknown>\n#1 0x562055a64c76 <unknown>\n#2 0x562055aa0c96 <unknown>\n#3 0x562055aa0dc1 <unknown>\n#4 0x562055ada7f4 <unknown>\n#5 0x562055ac003d <unknown>\n#6 0x562055ad830e <unknown>\n#7 0x562055abfde3 <unknown>\n#8 0x562055a952dd <unknown>\n#9 0x562055a9634e <unknown>\n#10 0x562055cf53e4 <unknown>\n#11 0x562055cf93d7 <unknown>\n#12 0x562055d03b20 <unknown>\n#13 0x562055cfa023 <unknown>\n#14 0x562055cc81aa <unknown>\n#15 0x562055d1e6b8 <unknown>\n#16 0x562055d1e847 <unknown>\n#17 0x562055d2e243 <unknown>\n#18 0x7f4d6a0bab43 <unknown>\n\nScreenshot: file:///home/runner/.selene/screenshots/1688982498799/1688982498799.png\nPageSource: file:///home/runner/.selene/screenshots/1688982498799/1688982498799.html","statusTrace":"fn = <selene.core.wait.Command object at 0x7fa164616f50>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/wait.py:119: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/wait.py:61: in __call__\n    return self._fn(entity)\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/entity.py:529: in raw_click\n    element.locate().click()\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/entity.py:189: in locate\n    return self._locator()\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/locator.py:34: in __call__\n    return self._locate()\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/entity.py:1132: in <lambda>\n    Locator(f'{self}.element({by})', lambda: self.driver.find_element(*by)),\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:740: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:346: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7fa1644fd990>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#16 0x562055d1e847 \\\\u003Cunknown>\\\\n#17 0x562055d2e243 \\\\u003Cunknown>\\\\n#18 0x7f4d6a0bab43 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        if status in ErrorCode.NO_SUCH_ELEMENT:\n            exception_class = NoSuchElementException\n        elif status in ErrorCode.NO_SUCH_FRAME:\n            exception_class = NoSuchFrameException\n        elif status in ErrorCode.NO_SUCH_SHADOW_ROOT:\n            exception_class = NoSuchShadowRootException\n        elif status in ErrorCode.NO_SUCH_WINDOW:\n            exception_class = NoSuchWindowException\n        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:\n            exception_class = StaleElementReferenceException\n        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:\n            exception_class = ElementNotVisibleException\n        elif status in ErrorCode.INVALID_ELEMENT_STATE:\n            exception_class = InvalidElementStateException\n        elif (\n            status in ErrorCode.INVALID_SELECTOR\n            or status in ErrorCode.INVALID_XPATH_SELECTOR\n            or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER\n        ):\n            exception_class = InvalidSelectorException\n        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:\n            exception_class = ElementNotSelectableException\n        elif status in ErrorCode.ELEMENT_NOT_INTERACTABLE:\n            exception_class = ElementNotInteractableException\n        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:\n            exception_class = InvalidCookieDomainException\n        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:\n            exception_class = UnableToSetCookieException\n        elif status in ErrorCode.TIMEOUT:\n            exception_class = TimeoutException\n        elif status in ErrorCode.SCRIPT_TIMEOUT:\n            exception_class = TimeoutException\n        elif status in ErrorCode.UNKNOWN_ERROR:\n            exception_class = WebDriverException\n        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:\n            exception_class = UnexpectedAlertPresentException\n        elif status in ErrorCode.NO_ALERT_OPEN:\n            exception_class = NoAlertPresentException\n        elif status in ErrorCode.IME_NOT_AVAILABLE:\n            exception_class = ImeNotAvailableException\n        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:\n            exception_class = ImeActivationFailedException\n        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:\n            exception_class = MoveTargetOutOfBoundsException\n        elif status in ErrorCode.JAVASCRIPT_ERROR:\n            exception_class = JavascriptException\n        elif status in ErrorCode.SESSION_NOT_CREATED:\n            exception_class = SessionNotCreatedException\n        elif status in ErrorCode.INVALID_ARGUMENT:\n            exception_class = InvalidArgumentException\n        elif status in ErrorCode.NO_SUCH_COOKIE:\n            exception_class = NoSuchCookieException\n        elif status in ErrorCode.UNABLE_TO_CAPTURE_SCREEN:\n            exception_class = ScreenshotException\n        elif status in ErrorCode.ELEMENT_CLICK_INTERCEPTED:\n            exception_class = ElementClickInterceptedException\n        elif status in ErrorCode.INSECURE_CERTIFICATE:\n            exception_class = InsecureCertificateException\n        elif status in ErrorCode.INVALID_COORDINATES:\n            exception_class = InvalidCoordinatesException\n        elif status in ErrorCode.INVALID_SESSION_ID:\n            exception_class = InvalidSessionIdException\n        elif status in ErrorCode.UNKNOWN_METHOD:\n            exception_class = UnknownMethodException\n        else:\n            exception_class = WebDriverException\n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"link text\",\"selector\":\"eroshenkoam/allure-example\"}\nE         (Session info: headless chrome=114.0.5735.198); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x562055d354e3 <unknown>\nE       #1 0x562055a64c76 <unknown>\nE       #2 0x562055aa0c96 <unknown>\nE       #3 0x562055aa0dc1 <unknown>\nE       #4 0x562055ada7f4 <unknown>\nE       #5 0x562055ac003d <unknown>\nE       #6 0x562055ad830e <unknown>\nE       #7 0x562055abfde3 <unknown>\nE       #8 0x562055a952dd <unknown>\nE       #9 0x562055a9634e <unknown>\nE       #10 0x562055cf53e4 <unknown>\nE       #11 0x562055cf93d7 <unknown>\nE       #12 0x562055d03b20 <unknown>\nE       #13 0x562055cfa023 <unknown>\nE       #14 0x562055cc81aa <unknown>\nE       #15 0x562055d1e6b8 <unknown>\nE       #16 0x562055d1e847 <unknown>\nE       #17 0x562055d2e243 <unknown>\nE       #18 0x7f4d6a0bab43 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:245: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\nbrowser_management = None\n\n    def test_github(browser_management):\n        browser.open('/')\n    \n        s('.header-search-button').click()\n        s('#query-builder-test').send_keys(\"eroshenkoam/allure-example\").press_enter()\n    \n>       s(by.link_text(\"eroshenkoam/allure-example\")).click()\n\ntests/test_selene.py:13: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/entity.py:533: in click\n    self.wait.for_(\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/wait.py:143: in for_\n    return self._decorator(self)(logic)(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nfn = <selene.core.wait.Command object at 0x7fa164616f50>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n                    reason_message = str(reason)\n    \n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=reason_message,\n                    )\n                    # TODO: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 4s, while waiting for:\nE                   browser.element(('link text', 'eroshenkoam/allure-example')).click\nE                   \nE                   Reason: NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"link text\",\"selector\":\"eroshenkoam/allure-example\"}\nE                     (Session info: headless chrome=114.0.5735.198); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE                   Stacktrace:\nE                   #0 0x562055d354e3 <unknown>\nE                   #1 0x562055a64c76 <unknown>\nE                   #2 0x562055aa0c96 <unknown>\nE                   #3 0x562055aa0dc1 <unknown>\nE                   #4 0x562055ada7f4 <unknown>\nE                   #5 0x562055ac003d <unknown>\nE                   #6 0x562055ad830e <unknown>\nE                   #7 0x562055abfde3 <unknown>\nE                   #8 0x562055a952dd <unknown>\nE                   #9 0x562055a9634e <unknown>\nE                   #10 0x562055cf53e4 <unknown>\nE                   #11 0x562055cf93d7 <unknown>\nE                   #12 0x562055d03b20 <unknown>\nE                   #13 0x562055cfa023 <unknown>\nE                   #14 0x562055cc81aa <unknown>\nE                   #15 0x562055d1e6b8 <unknown>\nE                   #16 0x562055d1e847 <unknown>\nE                   #17 0x562055d2e243 <unknown>\nE                   #18 0x7f4d6a0bab43 <unknown>\nE                   \nE                   Screenshot: file:///home/runner/.selene/screenshots/1688982498799/1688982498799.png\nE                   PageSource: file:///home/runner/.selene/screenshots/1688982498799/1688982498799.html\n\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selene/core/wait.py:141: TimeoutException","flaky":false,"newFailed":true,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"browser_management","time":{"start":1688982500824,"stop":1688982500824,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false}],"afterStages":[{"name":"browser_management::0","time":{"start":1688982507671,"stop":1688982507730,"duration":59},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false}],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_selene"},{"name":"host","value":"fv-az256-827"},{"name":"thread","value":"2388-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_selene"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":1,"broken":2,"skipped":0,"passed":1,"unknown":0,"total":4},"items":[{"uid":"92c122d0b9566dfd","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/8//#testresult/92c122d0b9566dfd","status":"passed","time":{"start":1688981953120,"stop":1688981960225,"duration":7105}},{"uid":"7ac388af4e5df7fc","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/7//#testresult/7ac388af4e5df7fc","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x56358b22f4e3 <unknown>\n#1 0x56358af5ec76 <unknown>\n#2 0x56358af87d78 <unknown>\n#3 0x56358af84029 <unknown>\n#4 0x56358afc2ccc <unknown>\n#5 0x56358afc247f <unknown>\n#6 0x56358afb9de3 <unknown>\n#7 0x56358af8f2dd <unknown>\n#8 0x56358af9034e <unknown>\n#9 0x56358b1ef3e4 <unknown>\n#10 0x56358b1f33d7 <unknown>\n#11 0x56358b1fdb20 <unknown>\n#12 0x56358b1f4023 <unknown>\n#13 0x56358b1c21aa <unknown>\n#14 0x56358b2186b8 <unknown>\n#15 0x56358b218847 <unknown>\n#16 0x56358b228243 <unknown>\n#17 0x7fbd97b3cb43 <unknown>","time":{"start":1688980172645,"stop":1688980173289,"duration":644}},{"uid":"4ff1187b2bae8427","reportUrl":"https://vapolonov.github.io/qa_guru_python_7/5//#testresult/4ff1187b2bae8427","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x55e1ca2a54e3 <unknown>\n#1 0x55e1c9fd4c76 <unknown>\n#2 0x55e1c9ffdd78 <unknown>\n#3 0x55e1c9ffa029 <unknown>\n#4 0x55e1ca038ccc <unknown>\n#5 0x55e1ca03847f <unknown>\n#6 0x55e1ca02fde3 <unknown>\n#7 0x55e1ca0052dd <unknown>\n#8 0x55e1ca00634e <unknown>\n#9 0x55e1ca2653e4 <unknown>\n#10 0x55e1ca2693d7 <unknown>\n#11 0x55e1ca273b20 <unknown>\n#12 0x55e1ca26a023 <unknown>\n#13 0x55e1ca2381aa <unknown>\n#14 0x55e1ca28e6b8 <unknown>\n#15 0x55e1ca28e847 <unknown>\n#16 0x55e1ca29e243 <unknown>\n#17 0x7f0f37c35b43 <unknown>","time":{"start":1688978997939,"stop":1688978998583,"duration":644}}]},"tags":[]},"source":"989b70b0033b5e6c.json","parameterValues":[]}